src/common/components/repositories-list/index.js

```diff
@@ -116,9 +117,21 @@ export class RepositoriesListView extends Component {
     );
   }
 
-  render() {
+  renderRows() {
     const { ids } = this.props.snaps;
 
+    if (this.props.hasSnaps) {
+      return Array.from(ids).sort().map(this.renderRow.bind(this));
+    } else if (this.props.snaps.isFetching && !this.props.snaps.success) {
+      // only show spinner during initial loading (not during polling for updates)
+      return <div className={styles.noRepos}><Spinner /></div>;
```

Comment:
We need to show the spinner in My repos on initial page load, so that we don't show "No repos" until we actually load snaps from server and know if user has any.